### PR TITLE
Deploy and Octopus workflows

### PR DESCRIPTION
**Pull request by alismx**
# PULL REQUEST

## Summary
- Adds a new deploy workflow that gets a tag, builds and pushes docker images and calls octopus for release and deployments.
- Some GH action version updates

## Related Issue
- resolves CDCgov/dibbs-cloud#46 (for now...)

## Additional Information
- Due to some unknowns in our remote environments downstream, I've made this manually triggered at the moment so that we don't have to watch failed jobs on merge or worry about it before we're ready.
- Trying to deploy `Staging` or `Production` should cause the job to exit, these environments are not yet available.

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+name: Deploy
+run-name: Deploy ${{github.event.inputs.branch }} to ${{ github.event.inputs.environment }} by @${{ github.actor }}
```
Comment (rin-skylight) [inline .github/workflows/deploy.yaml:3]: Non-blocking: inconsistent spacing for the variable.

File: .github/workflows/octopus.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+jobs:
+  octopus-release:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # Required for authentication
+    outputs:
+      release_number: ${{ steps.create_release.outputs.release_number }}
+    steps:
+      - name: Login to Octopus 🐙
```
Comment (rin-skylight) [inline .github/workflows/octopus.yaml:41]: <squee>! The emojis! Love this!

**Review by rin-skylight — APPROVED** [review]
Thank you for all your work on this!

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (alismx) [reply .github/workflows/deploy.yaml]: Fixed :heavy_check_mark:

File: .github/workflows/octopus.yaml (hunk 2 above)
Comment (alismx) [reply .github/workflows/octopus.yaml:41]: :octopus:

**Comment by alismx** [comment]
> @alismx this looks awesome! I'd like to apologize for the delayed review here, catching up after being OOO at a conference for most of last week.
> 
> I have one question. Maybe this is something I have missed, but I think we have eliminated the steps when we make a release that updates our API docs. The current process starts [on line 168](https://github.com/CDCgov/phdi/blob/main/.github/workflows/createNewRelease.yaml#L168) in `createNewRelease.yaml`.
> 
> We are slowly deprecating the Python SDK called `phdi` so updating those docs is no longer important, but the swagger docs for the APIs of all of the building blocks is are definitely something we want to maintain and continue to auto update.

@DanPaseltiner Thanks for taking a look! I understand what you're saying. I'll get that swagger doc process added back in.

**Comment by alismx** [comment]
Discussed with @DanPaseltiner , going to leave the current release workflow in place along with not fixing the broken doc generation steps as it's also broken on `main`. For now, if this workflow is triggered, docs will go to the `alis/tmp-docs` branch. The new workflow will be manual until more infrastructure is in place to require productionizing this process.

If I could get some reviews, I'd appreciate it! @DanPaseltiner @rin-skylight @marycrawford

**Pull request closed without merging**